src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -406,9 +406,8 @@ internal ReferenceTable
         internal void AddReference(AssemblyNameExtension assemblyName, Reference reference)
         {
             ErrorUtilities.VerifyThrow(assemblyName.Name != null, "Got an empty assembly name.");
-            if (References.ContainsKey(assemblyName))
+            if (References.TryGetValue(assemblyName, out Reference referenceGoingToBeReplaced))
```

Comment:
Is there a perf hit here?

Reply:
Actually this is a perf _gain_. Performing ContainsKey then getting the value performs the lookup twice, while TryGetValue performs the lookup once and sets the value to the out param.

Reply:
Yep! I don't expect it to be huge, but every little bit counts.